clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonApplicationLoadBalancerCachingAgent.groovy

```diff
@@ -95,25 +132,103 @@ class AmazonApplicationLoadBalancerCachingAgent extends AbstractAmazonLoadBalanc
       return null
     }
 
-    List<LoadBalancer> loadBalancers = metricsSupport.readData {
-      def loadBalancing = amazonClientProvider.getAmazonElasticLoadBalancingV2(account, region, true)
+    def loadBalancing = amazonClientProvider.getAmazonElasticLoadBalancingV2(account, region, true)
+
+    LoadBalancer loadBalancer = metricsSupport.readData {
       try {
         return loadBalancing.describeLoadBalancers(
           new DescribeLoadBalancersRequest().withNames([data.loadBalancerName as String])
-        ).loadBalancers
+        ).loadBalancers.get(0)
       } catch (LoadBalancerNotFoundException ignored) {
+        return null
+      }
+    }
+    if (loadBalancer == null) { loadBalancer = [:] }
```

Comment:
This seems wrong ...

`loadBalancer` is of type `LoadBalancer` and here you go assigning it to `[:]`.